ktor-server/ktor-server-plugins/ktor-server-single-page/jvm/src/io/ktor/plugins/SinglePageApplication.kt

```diff
@@ -0,0 +1,162 @@
+
+    /**
+     * Configuration type for the [SinglePageApplication] plugin
+     */
+    public class Configuration(
+        /**
+         * Default name of file or resource to serve when [applicationRoute] is requested
+         */
+        public var defaultPage: String = "index.html",
+
+        /**
+         * The URL path under which the content should be served
+         */
+        public var applicationRoute: String = "/",
+
+        /**
+         * Path under which the static content is located.
+         * Corresponds to the folder path if the [useResources] is true, resource path otherwise
+         */
+        public var filesPath: String = "",
+
+        /**
+         * Specifies if static content is a resource or folder
+         */
+        public var useResources: Boolean = false,
+
+        /**
+         * List of callbacks checking if file or resource in [filesPath] is ignored
+         * Request for such files or resources fails with 404 Forbidden status
+         */
+        internal val ignoredFiles: MutableList<(String) -> Boolean> = mutableListOf()
+    ) {
+        /**
+         * Register a [block] in [ignoredFiles]
+         * [block] returns true if [path] should be ignored.
+         */
+        public fun ignoreFiles(block: (path: String) -> Boolean) {
+            ignoredFiles += block
+        }
+
+        /**
+         * Creates a single page plugin application configuration for the Angular project.
+         * Resources will be shared from filesPath directory. Root file is index.html
+         */
+        public fun angular(filesPath: String) {
+            this.filesPath = filesPath
+        }
+
+        /**
+         * Creates a single page plugin application configuration for the React project.
+         * Resources will be shared from filesPath directory. Root file is index.html
+         */
+        public fun react(filesPath: String) {
+            this.filesPath = filesPath
+        }
+
+        /**
+         * Creates a single page plugin application configuration for the Vue project.
+         * Resources will be shared from filesPath directory. Root file is index.html
+         */
+        public fun vue(filesPath: String) {
+            this.filesPath = filesPath
+        }
+
+        /**
+         * Creates a single page plugin application configuration for the Ember project.
+         * Resources will be shared from filesPath directory. Root file is index.html
+         */
+        public fun ember(filesPath: String) {
+            this.filesPath = filesPath
+        }
+
+        /**
+         * Creates a single page plugin application configuration for the Backbone project.
+         * Resources will be shared from filesPath directory. Root file is index.html
+         */
+        public fun backbone(filesPath: String) {
+            this.filesPath = filesPath
+        }
+    }
+
+    /**
+     * Implementation of [ApplicationPlugin] for the [SinglePageApplication]
```

Comment:
What do you think if we remove [ApplicationPlugin] here? Suppose, a user presses F1 being at `install(SinglePageApplication)`, clicks `SinglePageApplication` (because there are no other distracting members) and reads the plugin description. Smth like this:

> The [SinglePageApplication] plugin implementation

>

Reply:
That's a good point, looks simplier. I will fix that